text/2020-03-07-copr-chunk.md

```diff
@@ -0,0 +1,150 @@
+# RFC: Using chunk format in coprocessor framework
+
+## Summary
+
+Using `chunk::Column` format in the `coprocessor` framework instead of `VectorValue`.
+
+The [chunk format](https://github.com/pingcap/tidb/blob/master/util/chunk/chunk.go) is a column based memory format and has much more compact memory layout than `VectorValue` with nullable values.
+
+## Motivation
+
+There are multiple benifits to use chunk format during operator evaluation:
+
+1. For many SQL builtin functions, they have the same null handling pattern. If some of input args are NULL, then the result is also NULL. We can call `merge_nulls` with input bitmap to output column before call the actual implementation. Then the actual function can iterate the column without null flag to reach better cache locality.
+2. SIMD instructions require continuous data in memory.
+3. Keep the same format with TiDB, so that it's possible to unify the computing framework between TiDB and TiKV.
+
```

Comment:
4. Using the Chunk format can improve performance for variable sized types like `Bytes`.